src/java/org/apache/cassandra/db/filter/ColumnFilter.java

```diff
@@ -201,40 +260,59 @@ public static ColumnFilter all(TableMetadata metadata)
      */
     public static ColumnFilter selection(RegularAndStaticColumns columns)
     {
-        return new ColumnFilter(false, false, (TableMetadata) null, columns, null);
+        return SelectionColumnFilter.newInstance(FetchingStrategy.ONLY_QUERIED_COLUMNS,
+                                                 (TableMetadata) null,
+                                                 columns,
+                                                 null);
```

Comment:
Nit: I think we don't need the cast of null, and without it this can be a one-liner:
```suggestion
        return SelectionColumnFilter.newInstance(FetchingStrategy.ONLY_QUERIED_COLUMNS, null, columns, null);
```